packages/lumx-react/src/hooks/useRovingTabIndex.tsx

```diff
@@ -0,0 +1,84 @@
+import get from 'lodash/get';
+import { RefObject, useEffect } from 'react';
+
+interface UseRovingTabIndexOptions {
+    parentRef: RefObject<HTMLElement>;
+    elementSelector: string;
+    keepTabIndex?: boolean;
+}
+
+export const useRovingTabIndex = ({ parentRef, elementSelector, keepTabIndex }: UseRovingTabIndexOptions) => {
+    useEffect(() => {
+        const parent = parentRef?.current;
+        if (!parent) {
+            return;
+        }
+
+        const elements = parent.querySelectorAll(elementSelector) as NodeListOf<HTMLElement>;
+        const initialFocusableElement = parent?.querySelector(`${elementSelector}[tabindex="0"]`);
+
+        const handleKeyDown = (index: number) => (evt: KeyboardEvent) => {
+            let newTabFocus = index;
+            if (!(evt.key === 'ArrowRight' || evt.key === 'ArrowLeft')) {
+                return;
+            }
+
+            let newElement: HTMLElement | undefined;
+            if (evt.key === 'ArrowRight') {
+                // Move right
+                while (!newElement || get(newElement, 'attributes.disabled')) {
```

Comment:
Was the while really necessary ? I'm not sure I see the gain from using it when all our targeted elements are already in a nodeList :/ 
Using a "while" seems like a good way of introducing an infinite loop IMO.

Also, disabled elements need to be focused when using the roving tabIndex technique, at least for tabs they should.
https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_disabled_controls
```
For the following composite widget elements, keep them focusable when disabled:
* Options in a Listbox
* Menu items in a Menu or menu bar
* Tab elements in a set of Tabs
* Tree items in a Tree View
```

Reply:
removing